test_board.py

```diff
@@ -74,8 +74,78 @@ def test_build_board(self):
+                               [6, 7, 8, 9, 10],
+                               [11, 12, 13, 14, 15],
+                               [16, 17, 18, 19, 20],
+                               [21, 22, 23, 24, 25]]
+            self.assertEqual(self.test_board.get_board(), expected_output)
+
+    def test_get_board_state(self):
+        self.test_board = Board(5)
+        with self.subTest('game is not finished if there are no winners'):
+            self.test_board._board = [[1, 2, 3, 4, 5],
+                                      [6, 7, 8, 9, 10],
+                                      [11, 12, 13, 14, 15],
+                                      [16, 17, 18, 19, 20],
+                                      [21, 22, 23, 24, 25]]
+            result = self.test_board.get_board_winner_status()
+            self.assertEqual(result, WinnerStatus.ONGOING)
+
+        with self.subTest('should check the state of the rows'):
+            self.test_board._board = [['x', 'x', 'x', 'x', 'x'],
+                                      [6, 7, 8, 9, 10],
+                                      [11, 12, 13, 14, 15],
+                                      [16, 17, 18, 19, 20],
+                                      [21, 22, 23, 24, 25]]
+            result = self.test_board.get_board_winner_status()
+            self.assertEqual(result, WinnerStatus.WINNER_X)
+
+        with self.subTest('should check the state of the columns'):
+            self.test_board._board = [['x', 2, 3, 4, 5],
+                                      ['x', 7, 8, 9, 10],
+                                      ['x', 12, 13, 14, 15],
+                                      ['x', 17, 18, 19, 20],
+                                      ['x', 22, 23, 24, 25]]
+            result = self.test_board.get_board_winner_status()
+            self.assertEqual(result, WinnerStatus.WINNER_X)
+
+        with self.subTest('should check diagonal descending'):
+            self.test_board._board = [['o', 2, 3, 4, 5],
+                                      [6, 'o', 8, 9, 10],
+                                      [11, 12, 'o', 14, 15],
+                                      [16, 17, 18, 'o', 20],
+                                      [21, 22, 23, 24, 'o']]
+            result = self.test_board.get_board_winner_status()
+            self.assertEqual(result, WinnerStatus.WINNER_O)
+
+        with self.subTest('should check diagonal ascending'):
+            self.test_board._board = [['o', 2, 3, 4, 'o'],
+                                      [6, 7, 8, 'o', 10],
+                                      [11, 12, 'o', 14, 15],
+                                      [16, 'o', 18, 19, 20],
+                                      ['o', 22, 23, 24, 25]]
+            result = self.test_board.get_board_winner_status()
+            self.assertEqual(result, WinnerStatus.WINNER_O)
+
+        with self.subTest('round is finished when tied'):
+            self.test_board._board = [['x', 'o', 'x', 'o', 'x'],
```

Comment:
This is a good name to me, it tells you what to expect (game is over) when (when there is a tie)